### PR TITLE
Test coverage further improvement

### PR DESCRIPTION
**Pull request by flickz**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

In continuation of the effort to increase the code coverage (#1335 ). 
- 1st commit: Having doubt about how I imported ../../lib/handleRequest again [here](https://github.com/flickz/fastify/blob/fix/improve-test-coverage/test/internals/handleRequest.test.js#L5) but couldn't figure out alternative way to do it. But let me know if you think there's another way I could have done it better.
- 2nd commit: Spent a lot of time trying to come with a test description :smile: but will really appreciate a better way to describe the test.

cc @Ethan-Arrowood

### CODE REVIEW

File: test/internals/handleRequest.test.js [hunk 1]
```diff
@@ -36,6 +37,14 @@ test('Request object', t => {
   t.strictDeepEqual(req.body, null)
 })
 
+test('handleRequest function - sent reply', t => {
```
Comment (mcollina) [inline test/internals/handleRequest.test.js]: which line is this testing?

File: test/wrapThenable.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const { kReplySentOverwritten } = require('../lib/symbols')
+const wrapThenable = require('../lib/wrapThenable')
+
+test('should resolve immediately when reply.sentOverwritten is true', t => {
```
Comment (mcollina) [inline test/wrapThenable.test.js]:
```suggestion
test('should resolve immediately when reply[kReplySentOverwritten] is true', t => {
```

File: test/wrapThenable.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  t.plan(1)
+  const reply = {}
+  reply[kReplySentOverwritten] = true
+  const thenable = Promise.resolve()
+  wrapThenable(thenable, reply)
+  t.pass()
```
Comment (mcollina) [inline test/wrapThenable.test.js]:
```suggestion
  t.end()
```

File: test/wrapThenable.test.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const { kReplySentOverwritten } = require('../lib/symbols')
+const wrapThenable = require('../lib/wrapThenable')
+
+test('should resolve immediately when reply.sentOverwritten is true', t => {
+  t.plan(1)
```
Comment (mcollina) [inline test/wrapThenable.test.js]:
```suggestion
```

File: test/wrapThenable.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  wrapThenable(thenable, reply)
+  t.pass()
+})
+
+test('should reject immediately when reply.sentOverwritten is true', t => {
+  t.plan(1)
+  const reply = { res: {} }
+  reply[kReplySentOverwritten] = true
+  reply.res.log = {
+    error: () => { }
```
Comment (mcollina) [inline test/wrapThenable.test.js]: can you add an assertion in here?

File: test/internals/handleRequest.test.js (hunk 1 above)
Comment (flickz) [reply test/internals/handleRequest.test.js]: This https://github.com/flickz/fastify/blob/139cbf40d1e4b6c0dae26701883071bc9af80439/lib/handleRequest.js#L9

File: test/wrapThenable.test.js (hunk 5 above)
Comment (flickz) [reply test/wrapThenable.test.js]: Yeah sure.

**Comment by Ethan-Arrowood** [comment]
Opened and reviewed all while I was asleep! Great work @flickz , and thank you @mcollina and @delvedor for reviewing 👍

**Pull request merged**